src/main/java/com/github/timothydowney/plugins/pipeline/npm/WithNPMStepExecution.java

```diff
@@ -123,22 +123,23 @@ public boolean start() throws Exception {
      * @throws AbortException in case of error
      */
     private void settingsFromConfig(String settingsConfigId, FilePath settingsFile) throws AbortException {
-
         Config config = ConfigFiles.getByIdOrNull(build, settingsConfigId);
         if (config == null) {
             throw new AbortException("Could not find the NPM config file id:" + settingsConfigId
                     + ". Make sure it exists on Managed Files");
         }
-        if (StringUtils.isBlank(config.content)) {
+
+        // Check if the content is blank and if authentication is not set, throw an exception
+        if (StringUtils.isBlank(config.content) && !isAuthenticationSet()) {
```

Comment:
To be honest I would simplify all of this and just remove the check. I think it must be the responsibility of the user to provide a valid file.

Also did you check the behavior of npm using an empty .npmrc file ?

Reply:
yes ,it's running fine . I have checked it with js-pipeline-utils pkg. After using .npmrc file

Reply:
I also think , it must be the responsibility of the user to provide a valid file, but that issue was open so worked on it